runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java

```diff
@@ -326,6 +324,17 @@ public DoFnOperator(
     this.finishBundleBeforeCheckpointing = flinkOptions.getFinishBundleBeforeCheckpointing();
   }
 
+  private boolean isRequiresStableInput(DoFn<InputT, OutputT> doFn) {
+    // WindowDoFnOperator does not use a DoFn
+    return doFn != null
```

Comment:
This annotation can be added to other parts of the DoFn like onTimer as per: https://github.com/apache/beam/blob/8dd87491dcb2660f859f87b3336adb984c62c12d/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFn.java#L803

Is `@ProcessElement` the only place we are trying to add support for it?

If you copied this code from https://github.com/apache/beam/blob/434427e90b55027c5944fa73de68bff4f9a4e8fe/runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/ParDoTranslation.java#L360, then that code is wrong based upon the specification.